### PR TITLE
Azure Storage Queue Track2

### PR DESCRIPTION
**Pull request by annatisch**

### CODE REVIEW

**Comment by adxsdk6** [comment]
Can one of the admins verify this patch?

File: sdk/storage/azure-storage-queue/azure/storage/queue/queue_service_client.py [hunk 1]
```diff
@@ -0,0 +1,426 @@
...
+    from .models import (
+        QueueProperties,
+        Logging,
+        Metrics,
+        CorsRule
+    )
+
+
+class QueueServiceClient(StorageAccountHostsMixin):
+    """A client interact with the Queue Service at the account level.
```
Comment (zezha-msft) [inline sdk/storage/azure-storage-queue/azure/storage/queue/queue_service_client.py]: typo

File: sdk/storage/azure-storage-queue/azure/storage/queue/queue_service_client.py [hunk 2]
```diff
@@ -0,0 +1,426 @@
...
+
+    This client provides operations to retrieve and configure the account properties
+    as well as list, create and delete queues within the account.
+    For operations relating to a specific queue, a client for this entity
+    can be retrieved using the :func:`~get_queue_client` function.
+
+    :param str account_url:
+        The URL to the queue storage account. Any other entities included
```
Comment (zezha-msft) [inline sdk/storage/azure-storage-queue/azure/storage/queue/queue_service_client.py]: minor: queue storage endpoint. storage account includes all 3 services.

File: sdk/storage/azure-storage-queue/azure/storage/queue/queue_client.py [hunk 3]
```diff
@@ -0,0 +1,768 @@
...
+from .models import QueueMessage, AccessPolicy, MessagesPaged
+
+if TYPE_CHECKING:
+    from datetime import datetime
+    from azure.core.pipeline.policies import HTTPPolicy
+    from .models import QueuePermissions, QueueProperties
+
+
+class QueueClient(StorageAccountHostsMixin):
+    """Creates a new QueueClient. This client represents interaction with a specific
```
Comment (zezha-msft) [inline sdk/storage/azure-storage-queue/azure/storage/queue/queue_client.py]: Grammar review needed.

File: sdk/storage/azure-storage-queue/azure/storage/queue/queue_client.py [hunk 4]
```diff
@@ -0,0 +1,768 @@
...
+    queue, although that queue may not yet exist.
+
+    :param str queue_url: The full URI to the queue. This can also be a URL to the storage
+        account, in which case the queue must also be specified.
+    :param queue: The queue. If specified, this value will override
+        a queue value specified in the blob URL.
```
Comment (zezha-msft) [inline sdk/storage/azure-storage-queue/azure/storage/queue/queue_client.py]: blob?

Comment (annatisch) [reply sdk/storage/azure-storage-queue/azure/storage/queue/queue_client.py]: Fixed in https://github.com/Azure/azure-sdk-for-python/pull/6186

File: sdk/storage/azure-storage-queue/azure/storage/queue/queue_service_client.py (hunk 2 above)
Comment (annatisch) [reply sdk/storage/azure-storage-queue/azure/storage/queue/queue_service_client.py]: Fixed in PR: https://github.com/Azure/azure-sdk-for-python/pull/6186

File: sdk/storage/azure-storage-queue/azure/storage/queue/queue_client.py (hunk 3 above)
Comment (annatisch) [reply sdk/storage/azure-storage-queue/azure/storage/queue/queue_client.py]: Fixed in PR: https://github.com/Azure/azure-sdk-for-python/pull/6186

File: sdk/storage/azure-storage-queue/azure/storage/queue/_queue_utils.py [hunk 5]
```diff
@@ -0,0 +1,170 @@
+# -------------------------------------------------------------------------
```
Comment (zezha-msft) [inline sdk/storage/azure-storage-queue/azure/storage/queue/_queue_utils.py]: consider refactoring this file into logically separated entities.

File: sdk/storage/azure-storage-queue/azure/storage/queue/queue_service_client.py (hunk 1 above)
Comment (annatisch) [reply sdk/storage/azure-storage-queue/azure/storage/queue/queue_service_client.py]: Fixed in PR: https://github.com/Azure/azure-sdk-for-python/pull/6186

File: sdk/storage/azure-storage-queue/azure/storage/queue/_queue_utils.py (hunk 5 above)
Comment (annatisch) [reply sdk/storage/azure-storage-queue/azure/storage/queue/_queue_utils.py]: Agreed - We have an issue open for this: https://github.com/Azure/azure-sdk-for-python/issues/6030

File: sdk/storage/azure-storage-queue/azure/storage/queue/_shared/constants.py [hunk 6]
```diff
@@ -0,0 +1,24 @@
+# -------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for
+# license information.
+# --------------------------------------------------------------------------
+
+import sys
+
+
+X_MS_VERSION = '2018-03-28'
```
Comment (bryevdv) [inline sdk/storage/azure-storage-queue/azure/storage/queue/_shared/constants.py]: This seems to duplicate  the version in https://github.com/Azure/azure-sdk-for-python/pull/6039/files#diff-8b893eaf830a6ba5ad078c41e67ba67b could there be a single source of truth for these? (Or are they only coincidentally identical at the moment)

Comment (annatisch) [reply sdk/storage/azure-storage-queue/azure/storage/queue/_shared/constants.py]: Agreed - I have opened an issue here:
https://github.com/Azure/azure-sdk-for-python/issues/6193

**Pull request merged**